### PR TITLE
better wording for the obsolete message

### PR DESCRIPTION
**Pull request by HEskandari**
Clarify the obsolete message for the `LogicalAddress`.

### CODE REVIEW

**Review by timbussmann — APPROVED** [review]

File: src/NServiceBus.Core.Tests/ApprovalFiles/APIApprovals.ApproveNServiceBus.approved.txt [hunk 1]
```diff
@@ -694,17 +694,20 @@ namespace NServiceBus
         [System.Obsolete("Use `QueueAddress.Qualifier` instead. The member currently throws a NotImplemente" +
             "dException. Will be removed in version 9.0.0.", true)]
         public string Qualifier { get; }
-        [System.Obsolete("Use `QueueAddress` instead. The member currently throws a NotImplementedException" +
-            ". Will be removed in version 9.0.0.", true)]
+        [System.Obsolete("Directly construct a QueueAddress with the discriminator. Use `QueueAddress` inst" +
+            "ead. The member currently throws a NotImplementedException. Will be removed in v" +
+            "ersion 9.0.0.", true)]
```
Comment (adamralph) [inline src/NServiceBus.Core.Tests/ApprovalFiles/APIApprovals.ApproveNServiceBus.approved.txt:699]: I'm not sure what the fix should be, or if a good one exists, but FWIW, this looks really weird:

> Directly construct a QueueAddress with the discriminator. Use `QueueAddress` instead.

Comment (adamralph) [reply src/NServiceBus.Core.Tests/ApprovalFiles/APIApprovals.ApproveNServiceBus.approved.txt:699]: Perhaps the `ReplacementTypeOrMember` parameter should be removed entirely from the attributes?

Comment (HEskandari) [reply src/NServiceBus.Core.Tests/ApprovalFiles/APIApprovals.ApproveNServiceBus.approved.txt:699]: @adamralph agreed. I'll remove the `ReplacementTypeOrMember`. I think it reads better that way.

Comment (HEskandari) [reply src/NServiceBus.Core.Tests/ApprovalFiles/APIApprovals.ApproveNServiceBus.approved.txt]: @adamralph updated.

Comment (adamralph) [reply src/NServiceBus.Core.Tests/ApprovalFiles/APIApprovals.ApproveNServiceBus.approved.txt]: 👍 LGTM.

**Pull request merged**